hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStoreFile.java

```diff
@@ -138,6 +140,12 @@ public class HStoreFile implements StoreFile {
   // Indicates if the file got compacted
   private volatile boolean compactedAway = false;
 
+  // Indicate if the file contains live cell versions for a given column
+  // in a row. MemStore flushes generate files with all cell versions. However,
+  // compactions can generate two files, one with the live cell versions and the other
+  // with the remaining (historical) cell versions.
+  private volatile boolean hasLiveVersions = true;
```

Comment:
Even when `hbase.hstore.defaultengine.enable.dualfilewriter` is enabled, the files generated by memstore flushes do not include the HAS_LIVE_VERSIONS HFile metadata key since memstore does not use compaction writers.  However, these files will include live version cells. When the HAS_LIVE_VERSIONS HFile metadata key is not included in the HFile, we still want to return true for HStoreFile#hasLiveVersion(). That is why the default is true. I will add more comments for this.